### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by JakobMiksch**
Functions shall throw errors when they fail. Before they returned `true`  or `false`

- There is an exception: If functions like `layer.get()`, `style.get()` oder `store.get()` do not find the object, because it does exist, then GeoServer returns undefined and does **not** throw an error. But there is a still an error thrown when GeoServer cannot be reached.
- adapted test suite to new approach
- in case of error the GeoServer output is added to the Error object using the property `geoServerOutput`, the normal "human-readable" message is stored in the standard `message` property. A failing request could be logged like this: `await grc.namespaces.delete(nsPrefix)`
```
#### Error message #####
Namespace doesn’t exist

#### GeoServer Response #####
<!doctype html><html lang="en"><head><title>HTTP Status 404 – Not Found</title><style type="text/css">h1 {font-family:Tahoma
,Arial,sans-serif;color:white;background-color:#525D76;font-size:22px;} h2 {font-family:Tahoma,Arial,sans-serif;color:white;
background-color:#525D76;font-size:16px;} h3 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-
size:14px;} body {font-family:Tahoma,Arial,sans-serif;color:black;background-color:white;} b {font-family:Tahoma,Arial,sans-
serif;color:white;background-color:#525D76;} p {font-family:Tahoma,Arial,sans-serif;background:white;color:black;font-size:1
2px;} a {color:black;} a.name {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h
1>HTTP Status 404 – Not Found</h1><hr class="line" /><p><b>Type</b> Status Report</p><p><b>Message</b> Namespace &#39;exampl
e-namespace&#39; not found</p><p><b>Description</b> The origin server did not find a current representation for the target r
esource or is not willing to disclose that one exists.</p><hr class="line" /><h3>Apache Tomcat/8.5.37</h3></body></html>
```

**TODO**

- [x] consider creating reusable error classes
- [x] adapt docs with `throws` statement
- [x] Update `demo` `index.js`
- [x] Add Upgrade Guide  --> an example is added to README
- [x] consider testing this new concept in an real application to ensure it works in production
- [x] fix indentation of `about.js`

### CODE REVIEW

**Comment by JakobMiksch** [comment]
@chrismayer I refined the error handling approach and made an extra handling for functions like `layer.get()`, `style.get()` oder `store.get()` . See the description in the PR text

**Comment by JakobMiksch** [comment]
@chrismayer this is ready to review again

File: README.md [hunk 1]
```diff
@@ -34,6 +34,22 @@ npm install
 npm run demo
 ```
 
+### Error Handling
+
+A request either succeeds or throws an Error. The thrown error has the standard `message` property with a "human-readable" text. Additionally the error has the property `geoServerOutput` which contains the direct response from GeoServer. This output is not guaranteed to exists and can be a simple text, or a complete HTML document. The latter is difficult to read, but might still be helpful for debugging. This example shows how these error properties can be used:
```
Comment (chrismayer) [inline README.md:39]:
```suggestion
A request either succeeds or throws an Error. The thrown error has the standard `message` property with a "human-readable" text. Additionally the error has the property `geoServerOutput` which contains the direct response from GeoServer. This output is not guaranteed to exist and can either be a simple text or a complete HTML document. The latter is difficult to read, but might still be helpful for debugging. This example shows how these error properties can be used:
```

File: README.md [hunk 2]
```diff
@@ -34,6 +34,22 @@ npm install
...
+
+```javascript
+  try {
+      // call any function from this library
+      await grc.styles.publish(workspace, styleName, sldBody)
+    } catch (error) {
+      // the standard error message
+      console.error(error.message);
+
+      // the whole Error including stack trace and (if available) the property 'geoServerOutput'
```
Comment (chrismayer) [inline README.md:49]:
```suggestion
      // the whole error including stack trace and (if available) the property 'geoServerOutput'
```

**Review by chrismayer — APPROVED** [review]
LGTM, thanks @JakobMiksch. This makes it really easier to handle errors.

Maybe after merging this we should add a hint that the master contains a breaking change to the README.

**Pull request merged**